diesel/src/query_builder/select_statement/dsl_impls.rs

```diff
@@ -196,3 +200,58 @@ impl<'a, S, F, D, W, O, L, Of, G, DB> InternalBoxedDsl<'a, DB>
         )
     }
 }
+
+impl<'a, F, D, W, O, L, Of, G, DB> InternalBoxedDsl<'a, DB>
```

Comment:
So I'm assuming I wrote this before you joined because we've not discussed this before. I'm curious on your thoughts though. I do write the types out explicitly [in the struct definition](https://github.com/diesel-rs/diesel/blob/6dec748273b29e95806ccc6a429c93dc4b2bf36d/diesel/src/query_builder/select_statement/mod.rs#L21-L30) because I think it's important to give them meaningful names. All of my usual principals say to use the full names, here, but this is what the code looks like.

```rust

impl<
    'a,
    Select,
    From,
    Distinct,
    Where,
    Order,
    Limit,
    Offset,
    Groupby,
    DB,
> InternalBoxedDsl<'a, DB>
    for SelectStatement<
        Select,
        From,
        Distinct,
        Where,
        Order,
        Limit,
        Offset,
        GroupBy,
    > where
        DB: Backend,
        Select: QueryFragment<DB> + SelectableExpression<From> + 'a,
        Distinct: QueryFragment<DB> + 'a,
        Where: Into<Option<Box<QueryFragment<DB> + 'a>>>,
        Order: QueryFragment<DB> + 'a,
        Limit: QueryFragment<DB> + 'a,
        Offset: QueryFragment<DB> + 'a,
{
    type Output = BoxedSelectStatement<'a, Select::SqlTypeForSelect, From, DB>;

    fn internal_into_boxed(self) -> Self::Output {
        BoxedSelectStatement::new(
            Box::new(self.select),
            self.from,
            Box::new(self.distinct),
            self.where_clause.into(),
            Box::new(self.order),
            Box::new(self.limit),
            Box::new(self.offset),
        )
    }
}
```